server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/method/EmailQueryMethod.scala

```diff
@@ -76,18 +95,31 @@ class EmailQueryMethod @Inject() (serializer: EmailQuerySerializer,
         limit = Some(limitToUse).filterNot(used => request.limit.map(_.value).contains(used.value))))
   }
 
-  private def searchQueryFromRequest(request: EmailQueryRequest): MultimailboxesSearchQuery = {
+  private def searchQueryFromRequest(request: EmailQueryRequest): Either[UnsupportedOperationException, MultimailboxesSearchQuery] = {
     val comparators: List[Comparator] = request.comparator.getOrElse(Set(Comparator.default)).toList
-    val sortedSearchQuery: SearchQuery = QueryFilter.buildQuery(request)
-      .sorts(comparators.map(_.toSort).asJava)
-      .build()
 
-    MailboxFilter.buildQuery(request, sortedSearchQuery)
+    comparators.map(_.toSort)
+      .sequence
+      .flatMap(sorts => for {
+        queryFilter <- QueryFilter.buildQuery(request)
+      } yield queryFilter
+        .sorts(sorts.asJava)
+        .build())
+      .map(MailboxFilter.buildQuery(request, _))
   }
 
   private def asEmailQueryRequest(arguments: Arguments): SMono[EmailQueryRequest] =
     serializer.deserializeEmailQueryRequest(arguments.value) match {
-      case JsSuccess(emailQueryRequest, _) => SMono.just(emailQueryRequest)
+      case JsSuccess(emailQueryRequest, _) => validateRequestParameters(emailQueryRequest)
       case errors: JsError => SMono.raiseError(new IllegalArgumentException(ResponseSerializer.serialize(errors).toString))
     }
+
+  private def validateRequestParameters(request: EmailQueryRequest): SMono[EmailQueryRequest] =
+    if (request.anchor.isDefined) {
+      SMono.raiseError(UnsupportedRequestParameterException("anchor"))
+    } else if (request.anchorOffset.isDefined) {
+      SMono.raiseError(UnsupportedRequestParameterException("anchorOffset"))
+    } else {
+      SMono.just(request)
+    }
```

Comment:
How about that: 
```
    if (request.anchor.isDefined) {
      SMono.raiseError(UnsupportedRequestParameterException("anchor"))
    }
    if (request.anchorOffset.isDefined) {
      SMono.raiseError(UnsupportedRequestParameterException("anchorOffset"))
    }
    SMono.just(request)
```

Reply:
Tried at first... if you do this you need to specify a return keyword for the above if statements, if not it will always return `SMono.just(request)`. So this is better I think

Reply:
ahh, i tried that in scala and we need keyword `return` but i dont know which one we preferred?

Reply:
we can considered that all agree with the current code.

Reply:
@nvduc91 in scala you should not use if without else.
I agree with the current code.
Could have been written with a pattern matching on the tuple (request.anchor, request.anchorOffset)

Reply:
Actually I like that suggestion thanks @remk :)